tfhe/src/integer/server_key/radix_parallel/tests_signed.rs

```diff
@@ -1335,6 +1472,121 @@ where
+        assert!(ct_res.block_carries_are_empty());
+        assert_eq!(ct_res, tmp_ct, "Failed determinism check");
+        assert_eq!(tmp_o, result_overflowed, "Failed determinism check");
+
+        let (expected_result, expected_overflowed) =
+            signed_overflowing_add_under_modulus(clear_0, clear_1, modulus);
+
+        let decrypted_result: i64 = cks.decrypt_signed(&ct_res);
+        let decrypted_overflowed = cks.decrypt_bool(&result_overflowed);
+        assert_eq!(
+            decrypted_result, expected_result,
+            "Invalid result for add, for ({clear_0} + {clear_1}) % {modulus} \
+             expected {expected_result}, got {decrypted_result}"
+        );
+        assert_eq!(
+            decrypted_overflowed,
+            expected_overflowed,
+            "Invalid overflow flag result for overflowing_suv for ({clear_0} + {clear_1}) % {modulus} \
+             expected overflow flag {expected_overflowed}, got {decrypted_overflowed}"
+        );
+
+        for _ in 0..NB_TEST_SMALLER {
+            // Add non zero scalar to have non clean ciphertexts
+            let clear_2 = random_non_zero_value(&mut rng, modulus);
+            let clear_3 = random_non_zero_value(&mut rng, modulus);
+
+            let ctxt_0 = sks.unchecked_scalar_add(&ctxt_0, clear_2);
+            let ctxt_1 = sks.unchecked_scalar_add(&ctxt_1, clear_3);
+
+            let clear_lhs = signed_add_under_modulus(clear_0, clear_2, modulus);
+            let clear_rhs = signed_add_under_modulus(clear_1, clear_3, modulus);
+
+            let d0: i64 = cks.decrypt_signed(&ctxt_0);
+            assert_eq!(d0, clear_lhs, "Failed sanity decryption check");
+            let d1: i64 = cks.decrypt_signed(&ctxt_1);
+            assert_eq!(d1, clear_rhs, "Failed sanity decryption check");
+
+            let (ct_res, result_overflowed) =
+                sks.signed_overflowing_add_parallelized(&ctxt_0, &ctxt_1);
+            assert!(ct_res.block_carries_are_empty());
+
+            let (expected_result, expected_overflowed) =
+                signed_overflowing_add_under_modulus(clear_lhs, clear_rhs, modulus);
+
+            let decrypted_result: i64 = cks.decrypt_signed(&ct_res);
+            let decrypted_overflowed = cks.decrypt_bool(&result_overflowed);
+            assert_eq!(
+                decrypted_result, expected_result,
+                "Invalid result for add, for ({clear_lhs} + {clear_rhs}) % {modulus} \
+                expected {expected_result}, got {decrypted_result}"
+            );
+            assert_eq!(
+                decrypted_overflowed,
+                expected_overflowed,
+                "Invalid overflow flag result for overflowing_add, for ({clear_lhs} + {clear_rhs}) % {modulus} \
+                expected overflow flag {expected_overflowed}, got {decrypted_overflowed}"
+            );
+        }
+    }
+
+    // Test with trivial inputs, as it was bugged at some point
+    for _ in 0..4 {
```

Comment:
no hard coded cases here ?

Reply:
hum no, we could add addition of trivial zero, but its an edge case like it was in subtractions due to the division